docker/sweepers_driver.py

```diff
@@ -58,11 +58,22 @@
 import json
 import logging
 import os
-from typing import Callable
+from typing import Callable, Iterable
 
 from pds.registrysweepers import provenance, ancestry
+from pds.registrysweepers.utils import configure_logging
+
+configure_logging(filepath=None, log_level=logging.INFO)
+log = logging.getLogger(__name__)
+
+dev_mode = str(os.environ.get("DEV_MODE")).lower() not in {'none', '', '0', 'false'}
+if dev_mode:
+    log.warning(f'Operating in development mode - host verification disabled')
+    import urllib3
+    urllib3.disable_warnings()
 
 opensearch_endpoint = os.environ.get("PROV_ENDPOINT")
+log.info(f'Targeting base OpenSearch endpoint "{opensearch_endpoint}"')
```

Comment:
When using Python logging, it's better to say `log.info('Blah blah %s to %s', thing1, thing2)` than to use string interpolation. If the `info` level is turned _off_, then no string substitutions are even executed.

Doing the string interpolation before the logging means it _always_ happens, even if that level is disabled.

Reply:
Agreed in principle, but seems like the kind of thing you that's only beneficial in performance-intensive contexts and outside of that I prefer to write for the human (readability is hardly an issue with a single substitution, but I still think it scans better).

Valid/not?

Reply:
[I guess this is the best-of-both-worlds answer](https://discuss.python.org/t/safer-logging-methods-for-f-strings-and-new-style-formatting/13802#big-bonus-switching-logging-to-the-modern-formatting-style-3)

Reply:
No doubt f-strings read 1000% better. But how can you ever predict that your "this'll never be called in a loop" won't be called in the future in some kind of huge or even massively-parallel loop?

Using the ugly `%` subs for logging is, sadly, the standard practice—unless you do this:

https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles

Reply:
Never mind, this is fine